lib/literature/uploaders.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Literature.Uploaders do
+  @moduledoc """
+    Literature Uploaders
+  """
+  use Waffle.Definition.Storage
+  use Waffle.Actions.Delete
+  use Literature.Uploaders.Actions.Store
+  use Literature.Uploaders.Actions.Url
+  use Literature.Uploaders.Versioning
+  use Waffle.Ecto.Definition
+
+  alias Literature.Config
+
+  @extension_whitelist ~w(.jpg .jpeg .png)
+  @versions ~w(original jpg webp avif)a
+
+  def asset_host, do: Config.waffle_asset_host()
+  def bucket, do: Config.waffle_bucket()
+  def __storage, do: Config.waffle_storage()
+
+  # Whitelist file extensions:
+  def validate({file, _}) do
+    Enum.member?(
+      @extension_whitelist,
+      file.file_name
+      |> Path.extname()
+      |> String.downcase()
+    )
+  end
+
+  def transform(:jpg, _) do
+    {:convert, "-format jpg", :jpg}
+  end
+
+  def transform(:webp, _) do
+    {:convert, "-format webp", :webp}
+  end
+
+  def transform(:avif, _) do
+    {:convert, "-format avif", :avif}
+  end
+
+  def storage_dir(_, _), do: "literature"
```

Comment:
@christianlj a quick look and it looks like it generates all the correct files sizes - we are just missing that each group of uploads is in its own directory to avoid file name conflicts.